src/components/MAPI/apps/AppValueConfigurator.tsx

```diff
@@ -0,0 +1,199 @@
+      return (
+        <>
+          User-level values are configured via the ConfigMap resource{' '}
+          <code>{configName}</code> in namespace <code>{configNamespace}</code>.
+        </>
+      );
+
+    case variant === AppValueConfiguratorVariant.ConfigMap && !hasConfig:
+      return 'No user-level values set';
+
+    case variant === AppValueConfiguratorVariant.Secret && hasConfig:
+      return (
+        <>
+          User-level secret values are configured via the Secret resource{' '}
+          <code>{configName}</code> in namespace <code>{configNamespace}</code>.
+        </>
+      );
+
+    case variant === AppValueConfiguratorVariant.Secret && !hasConfig:
+      return 'No user-level secret values set';
+
+    default:
+      return '';
+  }
+}
+
+function getSubtitleLabel(
+  variant: AppValueConfiguratorVariant,
+  configName: string,
+  configNamespace: string
+): React.ReactNode {
+  const hasConfig = isConfigValid(configName, configNamespace);
+
+  switch (true) {
+    case variant === AppValueConfiguratorVariant.ConfigMap && hasConfig:
+      return 'You can upload a YAML file here to update those values.';
+
+    case variant === AppValueConfiguratorVariant.ConfigMap && !hasConfig:
+      return 'You can upload a YAML file here to use as user-level values.';
+
+    case variant === AppValueConfiguratorVariant.Secret && hasConfig:
+      return 'You can upload a YAML file here to update those values.';
+
+    case variant === AppValueConfiguratorVariant.Secret && !hasConfig:
+      return 'You can upload a YAML file here to use as secret values.';
+
+    default:
+      return '';
+  }
+}
+
+function getButtonLabel(
+  variant: AppValueConfiguratorVariant,
+  configName: string,
+  configNamespace: string
+): string {
+  const hasConfig = isConfigValid(configName, configNamespace);
+
+  switch (true) {
+    case variant === AppValueConfiguratorVariant.ConfigMap && hasConfig:
+      return 'Replace values';
+
+    case variant === AppValueConfiguratorVariant.ConfigMap && !hasConfig:
+      return 'Upload values';
+
+    case variant === AppValueConfiguratorVariant.Secret && hasConfig:
+      return 'Replace secret values';
+
+    case variant === AppValueConfiguratorVariant.Secret && !hasConfig:
+      return 'Upload secret values';
+
+    default:
+      return '';
+  }
+}
+
+function getButtonIcon(
+  variant: AppValueConfiguratorVariant,
+  _configName: string,
+  _configNamespace: string
+): string {
+  if (variant === AppValueConfiguratorVariant.Secret) return 'fa fa-lock';
+
+  return 'fa fa-settings';
+}
+
+interface IAppValueConfiguratorProps
+  extends React.ComponentPropsWithoutRef<typeof Card> {
+  configName: string;
+  configNamespace: string;
+  variant?: AppValueConfiguratorVariant;
+  onUploadConfig?: (values: string) => Promise<void>;
+  onReplaceConfig?: (values: string) => Promise<void>;
+  isLoading?: boolean;
+}
+
+const AppValueConfigurator: React.FC<IAppValueConfiguratorProps> = ({
```

Comment:
Wasn't sure if this should be generic, and used for both configmaps and secrets, but it eventually worked out ok